src/Symmetry/LocalOperators.jl

```diff
@@ -0,0 +1,393 @@
+# spherical tensors, U' T_q U = D_qq′ T_q′, where D = exp(-i n⋅J), and J is a
+# spin operator in the spin-k representation. Observe that the standard
+# basis-convention for spin operators (eigenbasis of Jz, in descending order)
+# then determines the ordering of T_q and then 𝒪
+function stevens_abstract_polynomials(; J, k::Int)
+    k < 0  && error("Require k >= 0, received k=$k")
+    k > 6  && error("Stevens operators for k > 6 are currently unsupported, received k=$k.")
+
+    Jx, Jy, Jz = J
+    I = one(Jx)
+    X = Jx^2 + Jy^2 + Jz^2
+    Jp = Jx + im*Jy
+    Jm = Jx - im*Jy
+
+    A = [
+        [(1/2)  *(Jp^m + Jm^m) for m=k:-1:1]
+        [I];
+        [(1/2im)*(Jp^m - Jm^m) for m=1:k];
+    ]
+
+    B = if k == 0
+        [I]
+    elseif k == 1
+        [Jz,
+        I]
+    elseif k == 2
+        [3Jz^2 - X,
+        Jz,
+        I]
+    elseif k == 3
+        [5Jz^3-(3X-I)*Jz,
+        5Jz^2-X-I/2,
+        Jz,
+        I]
+    elseif k == 4
+        [35Jz^4 - (30X-25I)*Jz^2 + (3X^2-6X),
+        7Jz^3 - (3X+I)*Jz,
+        7Jz^2 - (X+5I),
+        Jz,
+        I]
+    elseif k == 5
+        [63Jz^5 - (70X-105I)*Jz^3 + (15X^2-50X+12I)*Jz,
+        21Jz^4 - 14X*Jz^2 + (X^2-X+(3/2)*I),
+        3Jz^3 - (X+6I)*Jz,
+        9Jz^2 - (X+(33/2)*I),
+        Jz,
+        I]
+    elseif k == 6
+        [231Jz^6 - (315X-735I)Jz^4 + (105X^2-525X+294I)*Jz^2 - (5X^3-40X^2+60X),
+        33Jz^5 - (30X-15I)*Jz^3 + (5X^2-10X+12I)*Jz,
+        33Jz^4 - (18X+123I)Jz^2 + (X^2+10X+102I),
+        11Jz^3 - (3X+59I)*Jz,
+        11Jz^2 - (X+38I),
+        Jz,
+        I]
+    elseif k > 6
+        # In principle, it should be possible to programmatically generate an
+        # arbitrary polynomial using Eq. (23) of I. D. Ryabov, J. Magnetic
+        # Resonance 140, 141-145 (1999), https://doi.org/10.1006/jmre.1999.1783
+        error("Stevens operators for k > 6 are currently unsupported, received k=$k.")
+    else # k < 0
+        error("Stevens operators require k >= 0, received k=$k")
+    end
+    B = [reverse(B); B[2:end]]
+
+    𝒪 = [(a*b+b*a)/2 for (a,b) = zip(A,B)]
+    return 𝒪
+end
+
+
+# Construct Stevens operators as polynomials in the spin operators.
+function stevens_matrices(N::Int, k::Int)
+    return stevens_abstract_polynomials(; J=gen_spin_ops(N), k)
+end
+
+
+# Construct Stevens operators in the classical limit, represented as polynomials
+# of spin expectation values
+function stevens_classical(k::Int)
+    𝒪s = stevens_abstract_polynomials(; J=spin_classical_symbols, k)
+    return map(𝒪s) do 𝒪
+        # In the large-S limit, only leading order terms contribute, yielding a
+        # homogeneous polynomial of degree k
+        𝒪 = sum(t for t in 𝒪 if DP.degree(t) == k)
+        # Remaining coefficients must be real integers; make this explicit
+        𝒪 = DP.mapcoefficients(x -> Int(x), 𝒪)
+        return 𝒪
+    end
+end
+
+# Construct explicit N-dimensional marix representation of operator
+function operator_to_matrix(p; N)
```

Comment:
This is not important, but I wonder if this should be exported. Certainly it's nice to keep everything in terms of abstract symbols, but I can imagine wanting to see the concrete result.

Reply:
Power users can always access this with `Sunny.operator_to_matrix()`, but I don't know if it's existence needs to be advertised to typical users. I could go either way though.